spec/controllers/videos_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+describe VideosController do
+  let(:user) { Fabricate(:user) }
+  let(:video) { Fabricate(:video) }
+
+  describe 'GET #index' do
+    it 'sets the @categories variable for authenticated users' do
+      request.session['user_id'] = user.id
+
+      local_video = video
+
+      get :index
+      expect(assigns(:categories)).to eq([local_video.category])
+    end
+
+    it 'redirects to root path for unauthenticated users' do
+      local_video = video
+
+      get :index
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe 'GET #show' do
+    context 'authenticated users' do
+      before { request.session['user_id'] = user.id }
+      before { Fabricate(:review, video: video) }
+      before do
+        get :show, id: video.id
+      end
+
+      it 'sets the @video variable' do
+        expect(assigns(:video)).to eq(video)
+      end
+
+      it 'sets the @reviews variable' do
+        expect(assigns(:reviews)).to be_present
+        expect(assigns(:reviews)).to match_array Review.all
```

Comment:
You don't need the first expectation as it this one cannot pass if the variable is not present.